src/pager/ui/view/render.rs

```diff
@@ -12,10 +12,24 @@ use crate::helpers;
 impl View {
     /// Render the view component
     pub fn render(
-        &self,
+        &mut self,
         stdout: &mut std::io::Stdout,
         lines: &Vec<String>,
     ) -> std::io::Result<Self> {
+        // Get the currently selected match
+        let selected = if self.search_matches.len() > 0 {
+            self.search_matches
+                .get(self.search_match_index % self.search_matches.len())
+                .unwrap_or(&(0 as u16, 0 as u16, 0 as u16))
+        } else {
+            &(0 as u16, 0 as u16, 0 as u16)
+        };
+
+        // If selected is out of view, scroll to it
+        if selected.0 > self.end() as u16 || selected.0 < self.start() as u16 {
+            self.scroll_row = (selected.0 as usize).saturating_sub(5);
```

Comment:
The `5` here represents the number of lines to leave as an offset when scrolling to position. But it is hard-coded.